server/controller/src/main/java/ai/starwhale/mlops/api/DatasetController.java

```diff
@@ -123,6 +126,35 @@ public ResponseEntity<ResponseMessage<DatasetInfoVo>> getDatasetInfo(String proj
         return ResponseEntity.ok(Code.success.asResponse(datasetInfo));
     }
 
+    @Override
+    public ResponseEntity<ResponseMessage<DataIndexDesc>> consumeNextData(String projectUrl,
+                                                                          String datasetUrl,
+                                                                          String versionUrl,
+                                                                          DataConsumptionRequest dataRangeRequest) {
+        var dataset = datasetService.query(projectUrl, datasetUrl, versionUrl);
+        var consumer = dataRangeRequest.getConsumerId().split("@");
+        if (consumer.length != 2) {
+            throw new SwValidationException(SwValidationException.ValidSubject.DATASET,
+                    "the format of consumerId validate error");
+        }
```

Comment:
`consumerID` join the parts in the client, and explodes in the server, why not just use two fields?

Reply:
it has been removed， replaced with timeout mechanism